docs/walletkit/ios/experimental/chain-abstraction.mdx

```diff
@@ -0,0 +1,141 @@
+# Chain Abstraction
+
+Chain Abstraction in WalletKit enables users with stablecoins on any network to spend them on-the-fly on a different network. Our Chain Abstraction solution provides a toolkit for wallet developers to integrate this complex functionality using WalletKit.
+
+For example, when an app requests a 100 USDC payment on Base network but the user only has USDC on Arbitrum, WalletKit offers methods to detect this mismatch, generate necessary transactions, track the cross-chain transfer, and complete the original transaction after bridging finishes.
+
+## How It Works
+
+:::note
+Apps need to pass `gas` as null, while sending a transaction to allow proper gas estimation by the wallet. Refer to this [guide](../../../appkit/next/experimental/chain-abstraction.mdx) for more details.
+:::
+
+When a app requests a transaction through `eth_sendTransaction`, you need to:
+1. Detect if chain abstraction is needed
+2. Generate necessary bridging transactions
+3. Handle transaction execution and monitoring
+4. Complete the original transaction once bridging is done
+
+The following sequence diagram illustrates the complete flow of a chain abstraction operation, from the initial dapp request to the final transaction confirmation
+
+![Chain Abstraction Flow](/assets/chain-abstraction-flow.png)
+
+## Methods
+
+Following are the methods from WalletKit that you will use in implementing chain abstraction.
+
+:::note
+💡 Chain abstraction is currently in experimental phase
+:::
+
+### Prepare 
+
+This method is used to check if chain abstraction is needed. If it is, it will return a response with the necessary transactions. 
+If it is not, it will return a response with the original transaction. 
+
+```swift
+@available(*, message: "This method is experimental. Use with caution.")
+public func prepare(transaction: InitialTransaction) async throws -> PrepareResponse {
+}
+```
+
+### Wait For Success With Timeout
+
+This method is used to wait for the all transactions to complete successfully. It will return a response with the transaction hash and receipt if the operation completed successfully. 
```

Comment:
```suggestion
This method is used to wait for the all bridging transactions to complete successfully. It will return a response with the transaction hash and receipt if the operation completed successfully. 
```